commit ca19dd154b15a7fe300d67d18dfc3c3465b12900 (refs/changes/72/772/1)
Author: Elijah Zupancic <elijah@zupancic.name>
Date:   2016-10-24T17:25:42-07:00 (2 years, 11 months ago)
    
    DOCKER-710 Need supported setup process/script for Windows docker clients to connect to Triton

diff --git a/tools/sdc-docker-setup.sh b/tools/sdc-docker-setup.sh
index 032ab69..38eeb61 100755
--- a/tools/sdc-docker-setup.sh
+++ b/tools/sdc-docker-setup.sh
@@ -509,7 +509,17 @@ caPath=$certDir/ca.pem
 
 mkdir -p $(dirname $keyPath)
 openssl rsa -in $sshPrivKeyPath -outform pem >$keyPath 2>/dev/null
-openssl req -new -key $keyPath -out $csrPath -subj "/CN=$account" >/dev/null 2>/dev/null
+
+# We proactively detect the Window's Ming shell and then we adapt to their
+# magic path syntax by adding a leading slash to the CN value. This allows for
+# the script to be run with the Git for Windows shell.
+cn="/CN=$account"
+if [[ -n "$(which uname)" && "`uname`" == MINGW* ]]; then
+    cn="/$cn"
+fi
+
+openssl req -new -key $keyPath -out $csrPath -subj "$cn" >/dev/null 2>/dev/null
+
 # TODO: expiry?
 openssl x509 -req -days 365 -in $csrPath -signkey $keyPath -out $certPath >/dev/null 2>/dev/null
 rm $csrPath    # The signing request has been used - remove it.
